app/Http/Controllers/Auth/DiscordController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Pterodactyl\Http\Controllers\Auth;
+
+use Exception;
+use Pterodactyl\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Auth\AuthManager;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Container\Container;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Http;
+use Pterodactyl\Http\Controllers\Controller;
+use Pterodactyl\Exceptions\DisplayException;
+use Pterodactyl\Services\Users\UserCreationService;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Contracts\Container\BindingResolutionException;
+
+class DiscordController extends Controller
+{
+    protected AuthManager $auth;
+    protected UserCreationService $creationService;
+
+    /**
+     * @throws BindingResolutionException
+     */
+    public function __construct(UserCreationService $creationService)
+    {
+        $this->auth = Container::getInstance()->make(AuthManager::class);
+        $this->creationService = $creationService;
+    }
+
+    /**
+     * @return JsonResponse
+     */
+    public function login(): JsonResponse
+    {
+        return new JsonResponse(['https://discord.com/api/oauth2/authorize?client_id='.env('DISCORD_CLIENT_ID').'&redirect_uri='.urlencode(env('REDIRECT_URL')).'&response_type=code&scope=identify%20email%20guilds%20guilds.join'], 200, [], null, true);
```

Comment:
Some issues with this system:

1) Instead of doing direct .env queries, please create a config file called `discord.php` in the `/config` directory. This will result in queries being fetched from the configuration instead of directly from the env, and also allows for catching errors (for example, if the variable in the env is invalid or doesn't exist outright).
2) Please format this correctly and store the variables locally. For example:
```$client = config('discord.client_id')```
3) Where possible, please use multiple lines and indent correctly - having the response as one long line is hard to read.